tasks/configuration.yml

```
@@ -1,79 +1,122 @@
 ---
 
+- name: populate timer facts
+  adfinis.facts.timer_facts:
+
+- name: populate service facts
+  ansible.builtin.service_facts:
+
 - name: create sync dirs
-  file:
+  ansible.builtin.file:
     path: "{{ repo_mirror_base_path }}/{{ item.name }}"
     state: directory
     mode: 0755
     owner: '{{ item.user | default(repo_mirror_user) }}'
     group: '{{ item.group | default(repo_mirror_group) }}'
   with_items: "{{ repo_mirror_repos }}"
-  when: item is defined and item
+  when: item.name is defined
   tags:
     - 'role::repo_mirror'
     - 'role::repo_mirror:configuration'
+    - 'role::repo_mirror:configuration:dirs'
 
 - name: create log dirs
-  file:
+  ansible.builtin.file:
     dest: "{{ repo_mirror_log_path }}/{{ item.name }}"
     state: directory
     mode: 0755
     owner: '{{ item.user | default(repo_mirror_user) }}'
     group: '{{ item.group | default(repo_mirror_group) }}'
   with_items: "{{ repo_mirror_repos }}"
-  when: item is defined and item
+  when: item.name is defined
   tags:
     - 'role::repo_mirror'
     - 'role::repo_mirror:configuration'
+    - 'role::repo_mirror:configuration:dirs'
 
 - name: create mirroring scripts
-  template:
+  ansible.builtin.template:
     src: "usr/local/bin/{{ item.type }}_mirror.j2"
     dest: "/usr/local/bin/mirror_{{ item.name }}.sh"
     owner: '{{ item.user | default(repo_mirror_user) }}'
     group: root
     mode: 0740
   with_items: "{{ repo_mirror_repos }}"
-  when: item is defined and item
+  when: item.name is defined
   tags:
     - 'role::repo_mirror'
     - 'role::repo_mirror:configuration'
-    - 'role::repo_mirror:configuration:script'
+    - 'role::repo_mirror:configuration:scripts'
 
 - name: create log rotation config
-  template:
+  ansible.builtin.template:
     src: etc/logrotate.d/config.template.j2
     dest: '/etc/logrotate.d/{{ item.name }}'
     owner: root
     group: root
     mode: 0744
-  with_items: '{{ repo_mirror_repos }}'
-  when: item is defined and item
+  with_items: "{{ repo_mirror_repos }}"
+  when: item.name is defined
   tags:
     - 'role::repo_mirror'
     - 'role::repo_mirror:configuration'
+    - 'role::repo_mirror:configuration:logrotate'
+
+- name: validate calendar format
+  ansible.builtin.command:
+    argv:
+      - "systemd-analyze"
+      - "calendar"
+      - "{{ item.systemd_timer_calendar }}"
+  register: systemd_analyze_output
```

```suggestion
  register: systemd_analyze_output
  changed_when: false
```
As this command does not change anything, it should report as such.